keys/keys.go

```diff
@@ -462,5 +463,26 @@ func Range(ba roachpb.BatchRequest) roachpb.RSpan {
 			to = endKey
 		}
 	}
+
+	verifyRange := func(ba roachpb.BatchRequest) error {
+		for _, union := range ba.Requests {
```

Comment:
please remove this loop; you're already iterating over the requests above. The verification can happen there, inline.
Also, please return an error from `keys.Range`.